oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/ReactiveOAuth2AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager.java

```diff
@@ -0,0 +1,136 @@
+	private final ReactiveOAuth2AuthorizedClientService reactiveAuthorizedClientService;
+	private ReactiveOAuth2AuthorizedClientProvider reactiveAuthorizedClientProvider = context -> null;
+	private Function<OAuth2AuthorizeRequest, Map<String, Object>> contextAttributesMapper = new DefaultContextAttributesMapper();
+
+	/**
+	 * Constructs an {@code ReactiveOAuth2AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager} using the provided parameters.
+	 *
+	 * @param reactiveClientRegistrationRepository the repository of client registrations
+	 * @param reactiveAuthorizedClientService      the authorized client service
+	 */
+	public ReactiveOAuth2AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager(ReactiveClientRegistrationRepository reactiveClientRegistrationRepository,
+			ReactiveOAuth2AuthorizedClientService reactiveAuthorizedClientService) {
+		Assert.notNull(reactiveClientRegistrationRepository, "reactiveClientRegistrationRepository cannot be null");
+		Assert.notNull(reactiveAuthorizedClientService, "reactiveAuthorizedClientService cannot be null");
+		this.reactiveClientRegistrationRepository = reactiveClientRegistrationRepository;
+		this.reactiveAuthorizedClientService = reactiveAuthorizedClientService;
+	}
+
+	@Nullable
+	@Override
+	public Mono<OAuth2AuthorizedClient> authorize(OAuth2AuthorizeRequest authorizeRequest) {
+		Assert.notNull(authorizeRequest, "authorizeRequest cannot be null");
+		String clientRegistrationId = authorizeRequest.getClientRegistrationId();
+		OAuth2AuthorizedClient authorizedClient = authorizeRequest.getAuthorizedClient();
+		Authentication principal = authorizeRequest.getPrincipal();
+		// @formatter:off
+		return Mono.justOrEmpty(authorizedClient)
+				.map(OAuth2AuthorizationContext::withAuthorizedClient)
+				.switchIfEmpty(Mono.defer(() -> this.reactiveClientRegistrationRepository.findByRegistrationId(clientRegistrationId)
+								.flatMap(clientRegistration -> this.reactiveAuthorizedClientService.loadAuthorizedClient(clientRegistrationId, principal.getName())
+										.map(OAuth2AuthorizationContext::withAuthorizedClient)
+										.switchIfEmpty(Mono.fromSupplier(() -> OAuth2AuthorizationContext.withClientRegistration(clientRegistration)))
+								)
+								.switchIfEmpty(Mono.error(new IllegalArgumentException("Could not find ClientRegistration with id '" + clientRegistrationId + "'")))
+							)
+				)
+				.map(contextBuilder -> contextBuilder.principal(principal)
+											.attributes(attributes -> {
+												Map<String, Object> contextAttributes = this.contextAttributesMapper.apply(authorizeRequest);
+												if (!CollectionUtils.isEmpty(contextAttributes)) {
+													attributes.putAll(contextAttributes);
+												}
+											})
+											.build()
+				).flatMap(authorizationContext -> this.reactiveAuthorizedClientProvider.authorize(authorizationContext)
+						.doOnNext(x -> reactiveAuthorizedClientService.saveAuthorizedClient(x, principal))
```

Comment:
Rename `x` to `authorizedClient`

Reply:
Renamed to _authorizedClient as conflicting with local variable authorizedClient.